lib/evmone/instructions.hpp

```diff
@@ -900,6 +900,29 @@ inline code_iterator swapn(StackTop stack, ExecutionState& state, code_iterator
     return pos + 2;
 }
 
+inline code_iterator mcopy(StackTop stack, ExecutionState& state, code_iterator pos) noexcept
```

Comment:
This code is based on `calldatacopy`, given the structure of this feature mirrors that.

The main difference is `std::max(dst_index, src_index)` ensures memory is expanded to the highest point, so copying can be sure it will copy 0s on padding. This matches the specification in the EIP.